### ISSUE TITLE
android: unable to restart tailscale 

### ISSUE THREAD

**Issue by rajveermalviya** [tailscale/tailscale#670]
**Describe the bug**
On Android tailscale only works on first startup. If I stop it from the app then I can't start it again.
Clearing the storage and logging in again, works. But that's clearly not ideal. 

**To Reproduce**
1. Install app
2. Login using Google, after this vpn will start automatically
3. Stop the vpn
4. Try starting again, switch in app turns on, but device vpn is not connected, also no notification is visible.

**Expected behavior**
It should start the vpn again.

**Screenshots**
https://youtu.be/jcj4-dTK244

**Version information:**
 - Device: Oneplus 7
 - OS: Android
 - OS version: Android 10
 - Tailscale version: v1.1.16

**Comment by eliasnaur** [comment]
Thank you for taking the time to record the bug. If you're comfortable with Android tools, please send me logs from `adb logcat` captured during the failing reconnect. If not, send me your Tailscale email address (to the address listed in my profile) and I may find something illuminating in the server logs. Make sure you keep the app open at least 2 minutes after the reconnect failure (logs are uploaded in batches to save power).

**Comment by rajveermalviya** [comment]
Alright so I now found out when it works and when it doesn't.
Here's a concise video: https://youtu.be/WA-hjBYFsqs

It works on first startup, but if I just press back and start the app again from the launcher
the switch does nothing. If I clear the app from memory and start it again then it works.

Here are the logs using `adb logcat | grep -i tailscale`

## When it works
### Turning switch on
```log
08-12 01:27:24.463 31087 31119 I com.tailscale.ipn: 19.6M/113.4M SetPrefs: Prefs{ra=true mesh=true dns=true want=true notepad=false derp=true shields=false routes=[] snat=true nf=on Persist{m=[uvlM9], o=[AAAAA], n=[Cv0AF] u="rajveer0malviya@gmail.com"}}
08-12 01:27:24.463 31087 31119 I com.tailscale.ipn: 19.6M/113.4M Switching ipn state Stopped -> Starting (WantRunning=true)
08-12 01:27:24.465 31087 31121 I com.tailscale.ipn: 19.6M/113.5M control: mapRoutine: unpaused
08-12 01:27:24.466 31087 31121 I com.tailscale.ipn: 19.6M/113.7M control: mapRoutine: state:authenticated
08-12 01:27:24.467 31087 31121 I com.tailscale.ipn: 19.6M/113.7M control: PollNetMap: stream=-1 :0 [219.91.241.129:19456 192.168.0.133:49380]
08-12 01:27:24.467 31087 31119 I com.tailscale.ipn: 19.6M/113.7M magicsock: SetPrivateKey called (init)
08-12 01:27:24.467 31087 31119 I com.tailscale.ipn: 19.6M/113.7M magicsock: private key changed, reconnecting to home derp-4
08-12 01:27:24.468 31087 31119 I com.tailscale.ipn: 19.6M/113.7M wgengine: Reconfig: configuring userspace wireguard config (with 3/3 peers)
08-12 01:27:24.468 31087 31119 I com.tailscale.ipn: 19.6M/113.7M device.Reconfig: resetting private key
08-12 01:27:24.474 31087 31119 I com.tailscale.ipn: 19.6M/113.7M UDP bind update skipped
08-12 01:27:24.475   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: startService() is called for servicecom.tailscale.ipn/.IPNService
08-12 01:27:24.475 31087 31133 I com.tailscale.ipn: 19.6M/113.7M device.Reconfig: new peer [ZFbNl]
08-12 01:27:24.475 31087 31150 I com.tailscale.ipn: 19.6M/113.7M magicsock: starting endpoint update (set-private-key)
08-12 01:27:24.475 31087 31150 I com.tailscale.ipn: 19.6M/113.7M magicsock: ReSTUN: endpoint update active, need another later ("non-zero-peers")
08-12 01:27:24.476 31087 31150 I com.tailscale.ipn: 19.6M/113.7M magicsock: adding connection to derp-4 for home-keep-alive
08-12 01:27:24.477 31087 31150 I com.tailscale.ipn: 19.6M/113.7M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
08-12 01:27:24.478 31087 31150 I com.tailscale.ipn: 19.8M/113.7M derphttp.Client.Connect: connecting to derp-4 (fra)
08-12 01:27:24.478 31087 31133 I com.tailscale.ipn: 19.8M/113.7M [ZFbNl] - Starting...
08-12 01:27:24.482 31087 31087 W m.tailscale.ipn: type=1400 audit(0.0:60404): avc: denied { read } for name="route" dev="proc" ino=4026532157 scontext=u:r:untrusted_app:s0:c73,c259,c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
08-12 01:27:24.480 31087 31133 I com.tailscale.ipn: 19.9M/113.7M device.Reconfig: new peer [URbEu]
08-12 01:27:24.481 31087 31133 I com.tailscale.ipn: 20.1M/113.7M [URbEu] - Starting...
08-12 01:27:24.482 31087 31133 I com.tailscale.ipn: 20.1M/113.7M magicsock: CreateEndpoint: key=[URbEu]: 8632c7a522486e5a9af94bea97007478e098b059e1e6de999f0a8200420d0472.disco.tailscale:12345
08-12 01:27:24.482 31087 31133 I com.tailscale.ipn: 20.1M/113.7M device.Reconfig: new peer [lv+6N]
08-12 01:27:24.484 31087 31133 I com.tailscale.ipn: 20.1M/113.7M [lv+6N] - Starting...
08-12 01:27:24.484 31087 31117 I com.tailscale.ipn: 20.1M/113.8M magicsock: CreateEndpoint: key=[lv+6N]: a816a3e90463e3ea38f13cf52bbd453879847e5220f43ae9309871c10c25b86f.disco.tailscale:12345
08-12 01:27:24.484 31087 31117 I com.tailscale.ipn: 20.1M/113.8M wgengine: Reconfig: configuring router
08-12 01:27:24.499   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: bindService is called for service : com.tailscale.ipn/.IPNService and for client system
08-12 01:27:24.499   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: total connections for service : com.tailscale.ipn/.IPNServiceare :1
08-12 01:27:24.505  1522 11049 I Vpn     : Established by com.tailscale.ipn on tun0
08-12 01:27:24.514 31087 31115 I com.tailscale.ipn: 20.3M/114.6M wgengine: Reconfig done
08-12 01:27:24.515 31087 31115 I com.tailscale.ipn: 20.3M/114.6M authReconfig: ra=true dns=true 0x0f: <nil>
08-12 01:27:24.515 31087 31144 I com.tailscale.ipn: 20.3M/114.6M Switching ipn state Starting -> Running (WantRunning=true)
08-12 01:27:24.535 31087 31150 I com.tailscale.ipn: interfaces: found Android default route 192.168.0.1
08-12 01:27:24.538   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: unbindService is called for service : com.tailscale.ipn/.IPNService and for client system
08-12 01:27:24.538   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: size of service connections for service: com.tailscale.ipn/.IPNServiceafter removal is 0
08-12 01:27:24.551 31087 31150 I com.tailscale.ipn: 20.7M/116.1M netcheck: probePortMapServices: me 192.168.0.133 -> gw 192.168.0.1
08-12 01:27:24.570   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: bindService is called for service : com.tailscale.ipn/.IPNService and for client system
08-12 01:27:24.570   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: total connections for service : com.tailscale.ipn/.IPNServiceare :1
08-12 01:27:24.573  1522  6490 I Vpn     : Established by com.tailscale.ipn on tun0
08-12 01:27:24.582  4486  4486 D TileServices: Couldn't find tile for ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.QuickToggleService}
08-12 01:27:24.601  6811  6811 D NotificationListener: onNotificationPosted# sbn: 238071354, package = com.tailscale.ipn
08-12 01:27:24.606 31087 31144 I com.tailscale.ipn: 21.3M/116.1M LinkChange(isExpensive=false); needsRebind=true
08-12 01:27:24.606 31087 31144 I com.tailscale.ipn: 21.3M/116.1M magicsock: link change, binding new port
08-12 01:27:24.618 31087 31144 I com.tailscale.ipn: 21.3M/116.1M control: mapRoutine: new map request during PollNetMap. canceling.
08-12 01:27:24.618 31087 31144 I com.tailscale.ipn: 21.3M/116.1M control: PollNetMap: context canceled
08-12 01:27:24.618 31087 31144 I com.tailscale.ipn: 21.3M/116.1M control: mapRoutine: state:authenticated
08-12 01:27:24.618 31087 31144 I com.tailscale.ipn: 21.3M/116.1M control: PollNetMap: stream=-1 :0 [219.91.241.129:19456 192.168.0.133:49380]
08-12 01:27:24.622  6811  6811 D NotificationListener: onNotificationPosted# sbn: 179047147, package = com.tailscale.ipn
08-12 01:27:24.625 31087 31115 I com.tailscale.ipn: 21.3M/116.1M magicsock: closing connection to derp-4 (rebind), age 0s
08-12 01:27:24.625 31087 31115 I com.tailscale.ipn: 21.3M/116.1M magicsock: 0 active derp conns
08-12 01:27:24.625 31087 31115 I com.tailscale.ipn: 21.3M/116.1M magicsock: ReSTUN: endpoint update active, need another later ("link-change-major")
08-12 01:27:24.625 31087 31115 I com.tailscale.ipn: 21.3M/116.1M magicsock: adding connection to derp-4 for home-keep-alive
08-12 01:27:24.625 31087 31115 I com.tailscale.ipn: 21.3M/116.1M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
08-12 01:27:24.626 31087 31115 I com.tailscale.ipn: 21.3M/116.1M derphttp.Client.Recv: connecting to derp-4 (fra)
08-12 01:27:24.668 31087 31144 I com.tailscale.ipn: 21.9M/116.1M LinkChange(isExpensive=false); needsRebind=false
08-12 01:27:24.668 31087 31144 I com.tailscale.ipn: 21.8M/116.1M magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
08-12 01:27:24.745 31087 31133 I com.tailscale.ipn: 21.9M/116.2M control: mapRoutine: netmap received: state:synchronized
08-12 01:27:24.745 31087 31133 I com.tailscale.ipn: 21.9M/116.2M control: sendStatus: mapRoutine2: state:synchronized
08-12 01:27:24.746 31087 31133 I com.tailscale.ipn: 21.9M/116.2M netmap diff: (none)
08-12 01:27:24.756 31087 31144 I com.tailscale.ipn: 22.5M/116.2M LinkChange(isExpensive=false); needsRebind=false
08-12 01:27:26.479 31087 31119 I com.tailscale.ipn: 22.6M/116.3M netcheck: udp=true v6=false mapvarydest=true hair=false portmap=UM v4a=219.91.241.129:19456 derp=4 derpdist=3v4:170ms,4v4:126ms
08-12 01:27:26.480 31087 31119 I com.tailscale.ipn: 22.6M/116.3M magicsock: netInfo update: NetInfo{varies=true hairpin=false ipv6=false udp=true derp=#4 portmap=UM link=""}
08-12 01:27:26.483 31087 31115 I com.tailscale.ipn: 22.6M/116.3M control: NetInfo: NetInfo{varies=true hairpin=false ipv6=false udp=true derp=#4 portmap=UM link=""}
08-12 01:27:26.484 31087 31115 I com.tailscale.ipn: 22.6M/116.3M control: cancelMapSafely: synced=true
08-12 01:27:26.486 31087 31115 I com.tailscale.ipn: 22.7M/116.3M control: PollNetMap: context canceled
08-12 01:27:26.487 31087 31115 I com.tailscale.ipn: 22.7M/116.3M control: mapRoutine: state:authenticated
08-12 01:27:26.487 31087 31115 I com.tailscale.ipn: 22.7M/116.3M control: PollNetMap: stream=-1 :0 [219.91.241.129:19456 192.168.0.133:49380]
08-12 01:27:26.493 31087 31144 I com.tailscale.ipn: 23.0M/116.3M magicsock: endpoints changed: 219.91.241.129:19456 (stun), 192.168.0.133:42577 (localAddresses)
08-12 01:27:26.494 31087 31144 I com.tailscale.ipn: 23.0M/116.3M control: client.newEndpoints(0, [219.91.241.129:19456 192.168.0.133:42577])
08-12 01:27:26.494 31087 31144 I com.tailscale.ipn: 23.0M/116.3M control: cancelMapSafely: synced=false
08-12 01:27:26.494 31087 31144 I com.tailscale.ipn: 23.0M/116.3M control: cancelMapSafely: wrote to channel
08-12 01:27:26.494 31087 31144 I com.tailscale.ipn: 23.0M/116.3M magicsock: starting endpoint update (link-change-minor)
08-12 01:27:26.529 31087 31144 I com.tailscale.ipn: interfaces: found Android default route 192.168.0.1
08-12 01:27:26.541 31087 31144 I com.tailscale.ipn: 24.4M/118.6M netcheck: probePortMapServices: me 192.168.0.133 -> gw 192.168.0.1
08-12 01:27:26.634 31087 31133 I com.tailscale.ipn: 24.5M/118.6M control: mapRoutine: new map request during PollNetMap. canceling.
08-12 01:27:26.636 31087 31133 I com.tailscale.ipn: 24.5M/118.6M control: PollNetMap: context canceled
08-12 01:27:26.636 31087 31133 I com.tailscale.ipn: 24.5M/118.6M control: mapRoutine: state:authenticated
08-12 01:27:26.637 31087 31133 I com.tailscale.ipn: 24.5M/118.6M control: PollNetMap: stream=-1 :0 [219.91.241.129:19456 192.168.0.133:42577]
08-12 01:27:26.782 31087 31133 I com.tailscale.ipn: 24.4M/118.6M control: mapRoutine: netmap received: state:synchronized
08-12 01:27:26.783 31087 31133 I com.tailscale.ipn: 24.4M/118.6M control: sendStatus: mapRoutine2: state:synchronized
08-12 01:27:26.783 31087 31133 I com.tailscale.ipn: 24.4M/118.6M netmap diff: (none)
08-12 01:27:26.923 31087 31144 I com.tailscale.ipn: 24.5M/118.8M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UM v4a=219.91.241.129:19475 derp=4 derpdist=1v4:201ms,3v4:172ms,4v4:140ms
08-12 01:27:26.924 31087 31144 I com.tailscale.ipn: 24.5M/118.8M magicsock: netInfo update: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#4 portmap=UM link=""}
08-12 01:27:26.925 31087 31115 I com.tailscale.ipn: 24.5M/118.8M control: NetInfo: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#4 portmap=UM link=""}
08-12 01:27:26.925 31087 31115 I com.tailscale.ipn: 24.5M/118.8M control: cancelMapSafely: synced=true
08-12 01:27:26.928 31087 31115 I com.tailscale.ipn: 24.5M/118.8M control: PollNetMap: context canceled
08-12 01:27:26.928 31087 31115 I com.tailscale.ipn: 24.5M/118.8M control: mapRoutine: state:authenticated
08-12 01:27:26.928 31087 31115 I com.tailscale.ipn: 24.6M/118.8M control: PollNetMap: stream=-1 :0 [219.91.241.129:19456 192.168.0.133:42577]
08-12 01:27:26.932 31087 31133 I com.tailscale.ipn: 24.7M/118.8M magicsock: endpoints changed: 219.91.241.129:19475 (stun), 192.168.0.133:42577 (localAddresses)
08-12 01:27:26.933 31087 31133 I com.tailscale.ipn: 24.7M/118.8M control: client.newEndpoints(0, [219.91.241.129:19475 192.168.0.133:42577])
08-12 01:27:26.934 31087 31133 I com.tailscale.ipn: 24.7M/118.8M control: cancelMapSafely: synced=false
08-12 01:27:26.934 31087 31133 I com.tailscale.ipn: 24.7M/118.8M control: cancelMapSafely: wrote to channel
08-12 01:27:27.070 31087 31119 I com.tailscale.ipn: 24.7M/118.8M control: mapRoutine: new map request during PollNetMap. canceling.
08-12 01:27:27.071 31087 31119 I com.tailscale.ipn: 24.7M/118.8M control: PollNetMap: context canceled
08-12 01:27:27.071 31087 31119 I com.tailscale.ipn: 24.7M/118.8M control: mapRoutine: state:authenticated
08-12 01:27:27.072 31087 31119 I com.tailscale.ipn: 24.7M/118.8M control: PollNetMap: stream=-1 :0 [219.91.241.129:19475 192.168.0.133:42577]
08-12 01:27:27.213 31087 31119 I com.tailscale.ipn: 24.8M/118.8M control: mapRoutine: netmap received: state:synchronized
08-12 01:27:27.213 31087 31119 I com.tailscale.ipn: 24.8M/118.8M control: sendStatus: mapRoutine2: state:synchronized
08-12 01:27:27.214 31087 31119 I com.tailscale.ipn: 24.8M/118.8M netmap diff: (none)
```

### Turning it off
```
08-12 01:27:46.172 31087 31115 I com.tailscale.ipn: 17.4M/121.6M SetPrefs: Prefs{ra=true mesh=true dns=true want=false notepad=false derp=true shields=false routes=[] snat=true nf=on Persist{m=[uvlM9], o=[AAAAA], n=[Cv0AF] u="rajveer0malviya@gmail.com"}}
08-12 01:27:46.173 31087 31115 I com.tailscale.ipn: 17.4M/121.6M Switching ipn state Running -> Stopped (WantRunning=false)
08-12 01:27:46.173 31087 31115 I com.tailscale.ipn: 17.4M/121.6M magicsock: SetPrivateKey called (zeroed)
08-12 01:27:46.174 31087 31115 I com.tailscale.ipn: 17.4M/121.6M magicsock: closing connection to derp-4 (zero-private-key), age 22s
08-12 01:27:46.174 31087 31115 I com.tailscale.ipn: 17.4M/121.6M magicsock: 0 active derp conns
08-12 01:27:46.176 31087 31150 I com.tailscale.ipn: 17.4M/121.6M magicsock: doing cleanup for discovery key a816a3e90463e3ea38f13cf52bbd453879847e5220f43ae9309871c10c25b86f
08-12 01:27:46.177 31087 31115 I com.tailscale.ipn: 17.3M/121.6M control: mapRoutine: paused
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M magicsock: doing cleanup for discovery key 5695402231f643b90e64bd66f286435fc185f7a5074d80a5ad5656d6c2403f05
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M magicsock: doing cleanup for discovery key 8632c7a522486e5a9af94bea97007478e098b059e1e6de999f0a8200420d0472
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M device.Reconfig: removing old peer [lv+6N]
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M [lv+6N] - Stopping...
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M control: mapRoutine: awaiting unpause
08-12 01:27:46.177 31087 31150 I com.tailscale.ipn: 17.3M/121.6M device.Reconfig: removing old peer [ZFbNl]
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M [ZFbNl] - Stopping...
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M device.Reconfig: removing old peer [URbEu]
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M [URbEu] - Stopping...
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M device.Reconfig: resetting private key
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M UDP bind update skipped
08-12 01:27:46.178 31087 31150 I com.tailscale.ipn: 17.3M/121.6M wgengine: Reconfig: configuring router
08-12 01:27:46.183   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: unbindService is called for service : com.tailscale.ipn/.IPNService and for client system
08-12 01:27:46.183   888  4887 I /vendor/bin/hw/vendor.qti.hardware.servicetracker@1.0-service: size of service connections for service: com.tailscale.ipn/.IPNServiceafter removal is 0
08-12 01:27:46.278 31087 31150 I com.tailscale.ipn: 17.3M/121.6M wgengine: Reconfig done
08-12 01:27:46.279 31087 31117 I com.tailscale.ipn: 17.7M/121.6M LinkChange(isExpensive=false); needsRebind=true
08-12 01:27:46.279 31087 31117 I com.tailscale.ipn: 17.7M/121.6M magicsock: link change, binding new port
08-12 01:27:46.272 31087 31087 W m.tailscale.ipn: type=1400 audit(0.0:60417): avc: denied { read } for name="route" dev="proc" ino=4026532157 scontext=u:r:untrusted_app:s0:c73,c259,c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
08-12 01:27:46.281  4486  4486 D TileServices: Couldn't find tile for ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.QuickToggleService}
08-12 01:27:46.286  6811  6811 D NotificationListener: onNotificationRemoved# sbn: 55996488, package = com.tailscale.ipn
08-12 01:27:46.291 31087 31144 I com.tailscale.ipn: 17.7M/121.7M magicsock: ReSTUN("link-change-major") ignored; stopped, no private key
```

## When it doesn't work
### Turning switch on
```
08-12 01:28:19.408 31087 31144 I com.tailscale.ipn: 19.8M/125.1M SetPrefs: Prefs{ra=true mesh=true dns=true want=true notepad=false derp=true shields=false routes=[] snat=true nf=on Persist{m=[uvlM9], o=[AAAAA], n=[Cv0AF] u="rajveer0malviya@gmail.com"}}
08-12 01:28:19.409 31087 31144 I com.tailscale.ipn: 19.8M/125.1M Switching ipn state Stopped -> Starting (WantRunning=true)
08-12 01:28:19.409 31087 31133 I com.tailscale.ipn: 19.8M/125.1M control: mapRoutine: unpaused
08-12 01:28:19.410 31087 31133 I com.tailscale.ipn: 19.8M/125.1M control: mapRoutine: state:authenticated
08-12 01:28:19.410 31087 31133 I com.tailscale.ipn: 19.8M/125.1M control: PollNetMap: stream=-1 :0 [219.91.241.129:19475 192.168.0.133:42577]
08-12 01:28:19.410 31087 31117 I com.tailscale.ipn: 19.8M/125.1M magicsock: SetPrivateKey called (init)
08-12 01:28:19.410 31087 31117 I com.tailscale.ipn: 19.8M/125.1M magicsock: private key changed, reconnecting to home derp-4
08-12 01:28:19.411 31087 31117 I com.tailscale.ipn: 19.8M/125.1M wgengine: Reconfig: configuring userspace wireguard config (with 3/3 peers)
08-12 01:28:19.411 31087 31117 I com.tailscale.ipn: 19.8M/125.1M device.Reconfig: resetting private key
08-12 01:28:19.412 31087 31150 I com.tailscale.ipn: 19.8M/125.1M magicsock: adding connection to derp-4 for home-keep-alive
08-12 01:28:19.412 31087 31150 I com.tailscale.ipn: 19.8M/125.1M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
08-12 01:28:19.412 31087 31150 I com.tailscale.ipn: 19.8M/125.1M magicsock: ReSTUN: endpoint update active, need another later ("non-zero-peers")
08-12 01:28:19.413 31087 31150 I com.tailscale.ipn: 19.8M/125.1M derphttp.Client.Recv: connecting to derp-4 (fra)
08-12 01:28:19.415 31087 31144 I com.tailscale.ipn: 19.8M/125.1M magicsock: starting endpoint update (set-private-key)
08-12 01:28:19.417 31087 31121 I com.tailscale.ipn: 20.1M/125.1M UDP bind update skipped
08-12 01:28:19.417 31087 31121 I com.tailscale.ipn: 20.1M/125.1M device.Reconfig: new peer [ZFbNl]
08-12 01:28:19.417 31087 31121 I com.tailscale.ipn: 20.1M/125.1M [ZFbNl] - Starting...
08-12 01:28:19.412 31087 31087 W m.tailscale.ipn: type=1400 audit(0.0:60421): avc: denied { read } for name="route" dev="proc" ino=4026532157 scontext=u:r:untrusted_app:s0:c73,c259,c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
08-12 01:28:19.418 31087 31121 I com.tailscale.ipn: 20.1M/125.1M magicsock: CreateEndpoint: key=[ZFbNl]: 5695402231f643b90e64bd66f286435fc185f7a5074d80a5ad5656d6c2403f05.disco.tailscale:12345
08-12 01:28:19.419 31087 31121 I com.tailscale.ipn: 20.2M/125.1M device.Reconfig: new peer [URbEu]
08-12 01:28:19.419 31087 31121 I com.tailscale.ipn: 20.2M/125.1M [URbEu] - Starting...
08-12 01:28:19.420 31087 31121 I com.tailscale.ipn: 20.2M/125.1M magicsock: CreateEndpoint: key=[URbEu]: 8632c7a522486e5a9af94bea97007478e098b059e1e6de999f0a8200420d0472.disco.tailscale:12345
08-12 01:28:19.420 31087 31121 I com.tailscale.ipn: 20.2M/125.1M device.Reconfig: new peer [lv+6N]
08-12 01:28:19.420 31087 31121 I com.tailscale.ipn: 20.2M/125.1M [lv+6N] - Starting...
08-12 01:28:19.421 31087 31121 I com.tailscale.ipn: 20.2M/125.1M magicsock: CreateEndpoint: key=[lv+6N]: a816a3e90463e3ea38f13cf52bbd453879847e5220f43ae9309871c10c25b86f.disco.tailscale:12345
08-12 01:28:19.421 31087 31121 I com.tailscale.ipn: 20.2M/125.1M wgengine: Reconfig: configuring router
08-12 01:28:19.421 31087 31121 I com.tailscale.ipn: 20.2M/125.1M wgengine: Reconfig done
08-12 01:28:19.421 31087 31121 I com.tailscale.ipn: 20.2M/125.1M authReconfig: ra=true dns=true 0x0f: <nil>
08-12 01:28:19.422 31087 31121 I com.tailscale.ipn: 20.2M/125.1M Switching ipn state Starting -> Running (WantRunning=true)
08-12 01:28:19.424  4486  4486 D TileServices: Couldn't find tile for ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.QuickToggleService}
08-12 01:28:19.446 31087 31117 I com.tailscale.ipn: interfaces: found Android default route 192.168.0.1
08-12 01:28:19.455 31087 31117 I com.tailscale.ipn: 20.7M/125.1M netcheck: probePortMapServices: me 192.168.0.133 -> gw 192.168.0.1
08-12 01:28:19.452 31087 31087 W m.tailscale.ipn: type=1400 audit(0.0:60423): avc: denied { read } for name="route" dev="proc" ino=4026532157 scontext=u:r:untrusted_app:s0:c73,c259,c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
08-12 01:28:19.572 31087 31150 I com.tailscale.ipn: 20.7M/125.4M control: mapRoutine: netmap received: state:synchronized
08-12 01:28:19.572 31087 31150 I com.tailscale.ipn: 20.7M/125.4M control: sendStatus: mapRoutine2: state:synchronized
08-12 01:28:19.572 31087 31150 I com.tailscale.ipn: 20.7M/125.4M netmap diff: (none)
08-12 01:28:19.867 31087 31150 I com.tailscale.ipn: 20.8M/125.6M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UM v4a=219.91.241.129:19512 derp=4 derpdist=1v4:214ms,3v4:174ms,4v4:138ms
08-12 01:28:19.871 31087 31150 I com.tailscale.ipn: 21.0M/125.6M magicsock: endpoints changed: 219.91.241.129:19512 (stun), 192.168.0.133:41577 (localAddresses)
08-12 01:28:19.872 31087 31121 I com.tailscale.ipn: 21.0M/125.6M control: client.newEndpoints(0, [219.91.241.129:19512 192.168.0.133:41577])
08-12 01:28:19.872 31087 31121 I com.tailscale.ipn: 21.0M/125.6M control: cancelMapSafely: synced=true
08-12 01:28:19.874 31087 31121 I com.tailscale.ipn: 21.0M/125.6M magicsock: starting endpoint update (non-zero-peers)
08-12 01:28:19.874 31087 31150 I com.tailscale.ipn: 21.0M/125.6M control: PollNetMap: context canceled
08-12 01:28:19.875 31087 31150 I com.tailscale.ipn: 21.0M/125.6M control: mapRoutine: state:authenticated
08-12 01:28:19.875 31087 31150 I com.tailscale.ipn: 21.0M/125.6M control: PollNetMap: stream=-1 :0 [219.91.241.129:19512 192.168.0.133:41577]
08-12 01:28:19.872 31087 31087 W m.tailscale.ipn: type=1400 audit(0.0:60424): avc: denied { read } for name="route" dev="proc" ino=4026532157 scontext=u:r:untrusted_app:s0:c73,c259,c512,c768 tcontext=u:object_r:proc_net:s0 tclass=file permissive=0
08-12 01:28:19.914 31087 31150 I com.tailscale.ipn: interfaces: found Android default route 192.168.0.1
08-12 01:28:19.925 31087 31150 I com.tailscale.ipn: 22.0M/125.6M netcheck: probePortMapServices: me 192.168.0.133 -> gw 192.168.0.1
08-12 01:28:20.011 31087 31121 I com.tailscale.ipn: 22.0M/125.6M control: mapRoutine: netmap received: state:synchronized
08-12 01:28:20.011 31087 31121 I com.tailscale.ipn: 22.0M/125.6M control: sendStatus: mapRoutine2: state:synchronized
08-12 01:28:20.012 31087 31121 I com.tailscale.ipn: 22.0M/125.6M netmap diff: (none)
08-12 01:28:20.335 31087 31121 I com.tailscale.ipn: 22.0M/125.9M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UM v4a=219.91.241.129:19512 derp=4 derpdist=1v4:217ms,3v4:176ms,4v4:137ms
```

### Turning it off
```08-12 01:28:35.376 31087 31121 I com.tailscale.ipn: 22.9M/126.6M SetPrefs: Prefs{ra=true mesh=true dns=true want=false notepad=false derp=true shields=false routes=[] snat=true nf=on Persist{m=[uvlM9], o=[AAAAA], n=[Cv0AF] u="rajveer0malviya@gmail.com"}}
08-12 01:28:35.376 31087 31121 I com.tailscale.ipn: 22.9M/126.6M Switching ipn state Running -> Stopped (WantRunning=false)
08-12 01:28:35.377 31087 31121 I com.tailscale.ipn: 22.9M/126.7M magicsock: SetPrivateKey called (zeroed)
08-12 01:28:35.377 31087 31121 I com.tailscale.ipn: 22.9M/126.7M magicsock: closing connection to derp-4 (zero-private-key), age 16s
08-12 01:28:35.377 31087 31121 I com.tailscale.ipn: 22.9M/126.7M magicsock: 0 active derp conns
08-12 01:28:35.378 31087 31121 I com.tailscale.ipn: 22.9M/126.7M magicsock: doing cleanup for discovery key 8632c7a522486e5a9af94bea97007478e098b059e1e6de999f0a8200420d0472
08-12 01:28:35.378 31087 31117 I com.tailscale.ipn: 22.9M/126.7M control: mapRoutine: paused
08-12 01:28:35.378 31087 31119 I com.tailscale.ipn: 22.9M/126.7M magicsock: doing cleanup for discovery key a816a3e90463e3ea38f13cf52bbd453879847e5220f43ae9309871c10c25b86f
08-12 01:28:35.380 31087 31119 I com.tailscale.ipn: 22.9M/126.7M magicsock: doing cleanup for discovery key 5695402231f643b90e64bd66f286435fc185f7a5074d80a5ad5656d6c2403f05
08-12 01:28:35.381 31087 31119 I com.tailscale.ipn: 22.9M/126.7M wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
08-12 01:28:35.381 31087 31119 I com.tailscale.ipn: 22.9M/126.7M device.Reconfig: removing old peer [ZFbNl]
08-12 01:28:35.382 31087 31119 I com.tailscale.ipn: 22.9M/126.7M [ZFbNl] - Stopping...
08-12 01:28:35.382 31087 31117 I com.tailscale.ipn: 22.9M/126.7M control: mapRoutine: awaiting unpause
08-12 01:28:35.382 31087 31144 I com.tailscale.ipn: 22.9M/126.7M device.Reconfig: removing old peer [URbEu]
08-12 01:28:35.383 31087 31144 I com.tailscale.ipn: 22.9M/126.7M [URbEu] - Stopping...
08-12 01:28:35.383 31087 31147 I com.tailscale.ipn: 22.9M/126.7M device.Reconfig: removing old peer [lv+6N]
08-12 01:28:35.384 31087 31147 I com.tailscale.ipn: 22.9M/126.7M [lv+6N] - Stopping...
08-12 01:28:35.384 31087 31117 I com.tailscale.ipn: 22.9M/126.7M device.Reconfig: resetting private key
08-12 01:28:35.384 31087 31117 I com.tailscale.ipn: 22.9M/126.7M UDP bind update skipped
08-12 01:28:35.384 31087 31117 I com.tailscale.ipn: 22.9M/126.7M wgengine: Reconfig: configuring router
08-12 01:28:35.384 31087 31117 I com.tailscale.ipn: 22.9M/126.7M wgengine: Reconfig done
08-12 01:28:35.387  4486  4486 D TileServices: Couldn't find tile for ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.QuickToggleService}
```

**Comment by kellytrinh** [comment]
Just looking at that youtube - would be nice to have a feature to name the devices?  Three called 'Oneplus 7' certainly is hard to handle (I have much the same issue with  my setup in-so-far as the defaults aren't particularly great)

**Comment by eliasnaur** [comment]
@rajveermalviya I just released version 0.1.18 to the open beta track. Please let me know whether that fixes your issue.

You can join the beta test track through this link: https://play.google.com/store/apps/details?id=com.tailscale.ipn. Or wait a day or two until I release 0.1.18 to all users.

**Comment by rajveermalviya** [comment]
apologies for late reply, but yep the latest update fixes the issue.
